ledger.go

```diff
@@ -738,100 +738,88 @@ func (l *Ledger) finalize(block Block) {
 func (l *Ledger) query() {
 	snowballK := conf.GetSnowballK()
 
-	if len(l.client.ClosestPeerIDs()) < snowballK {
-		// TODO not enough peers, what do to ?
+	peers, err := SelectPeers(l.client.ClosestPeers(), snowballK)
+	if err != nil {
 		return
 	}
 
 	current := l.blocks.Latest()
 
-	workerChan := make(chan *grpc.ClientConn, 16)
-
-	var workerWG sync.WaitGroup
-	workerWG.Add(cap(workerChan))
+	var wg sync.WaitGroup
 
 	voteChan := make(chan *finalizationVote, snowballK)
 
 	req := &QueryRequest{BlockIndex: current.Index + 1}
 
-	for i := 0; i < cap(workerChan); i++ {
-		go func() {
-			for conn := range workerChan {
-				f := func() {
-					client := NewWaveletClient(conn)
+	for _, p := range peers {
+		wg.Add(1)
```

Comment:
wg.Add(len(peers))

Reply:
there no difference, but i just prefer to keep `Add` and `Done` close to each other